cli/src/commands/commit.rs

```diff
@@ -80,6 +78,16 @@ new working-copy commit.
         )?;
     }
 
+    let template = description_template_for_commit(
+        ui,
+        command.settings(),
+        tx.base_workspace_helper(),
+        "",
+        commit.description(),
+        &base_tree,
+        &middle_tree,
+    )?;
+
```

Comment:
The code looks good, thanks! Could you also add a test? It looks like you could copy and/or modify one of these two: https://github.com/martinvonz/jj/blob/b11f7eee1b5b43f4589cb177d1bcbe4438ace17e/cli/tests/test_commit_command.rs#L36-L108

Reply:
Sure, I have no idea how to make the interactive selection work though, should it be something like this:

```Rust
#[test]
fn test_commit_interactive() {
    let mut test_env = TestEnvironment::default();
    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
    let workspace_path = test_env.env_root().join("repo");

    std::fs::write(workspace_path.join("file1"), "foo\n").unwrap();
    std::fs::write(workspace_path.join("file2"), "bar\n").unwrap();
    let edit_script = test_env.set_up_fake_editor();
    std::fs::write(&edit_script, ["dump editor"].join("\0")).unwrap();

    let diff_editor = test_env.set_up_fake_diff_editor();
    std::fs::write(&diff_editor, "reset file2").unwrap();

    // Create a commit interactively and select only file1
    test_env.jj_cmd_ok(&workspace_path, &["commit", "-i"]);

    insta::assert_snapshot!(
        std::fs::read_to_string(test_env.env_root().join("editor")).unwrap(), @r###"

    JJ: This commit contains the following changes:
    JJ:     A file1

    JJ: Lines starting with "JJ: " (like this one) will be removed.
    "###);
}
```

Reply:
Ah, you figured out how to do it :) I didn't notice comments below until now. Feel free to merge!